File: Web-Frontend/src/api/taskApi.js
import API from "./axios";



export const getTasks = (params) => API.get("/tasks", { params });

export const getTaskById = (id) => API.get(`/tasks/${id}`);

export const getTaskStats = () => API.get("/tasks/stats");

export const createTask = (data) => API.post("/tasks", data);

export const updateTask = (id, formData) =>
  API.put(`/tasks/${id}`, formData, {
    headers: { "Content-Type": "multipart/form-data" },
  });

export const updateTaskStatus = (id, status, comment = "") =>{
  const fd = new FormData();
  fd.append("status", status);
  if (comment) fd.append("comment", comment);
  return API.put(`/tasks/${id}`, fd, {
    headers: { "Content-Type": "multipart/form-data" },
  });
};

export const deleteTask = (id) => API.delete(`/tasks/${id}`);

File: Mobile-App/mint_mobile/lib/api/task_api.dart
import 'package:dio/dio.dart';
import 'dio_client.dart';

class TaskApi {
  static Future<Response> getTasks() => DioClient.instance.get('/tasks');

  static Future<Response> getTaskById(String id) =>
      DioClient.instance.get('/tasks/$id');

  // FIX: send JSON instead of FormData so backend req.body can read it
  static Future<Response> createTask(Map<String, dynamic> data) =>
      DioClient.instance.post(
        '/tasks',
        data: data,
        options: Options(contentType: 'application/json'),
      );

  // Keep as FormData — handles file uploads
  static Future<Response> updateTask(String id, FormData formData) =>
      DioClient.instance.put(
        '/tasks/$id',
        data: formData,
        options: Options(contentType: 'multipart/form-data'),
      );

  static Future<Response> deleteTask(String id) =>
      DioClient.instance.delete('/tasks/$id');

  static Future<Response> getTaskStats() =>
      DioClient.instance.get('/tasks/stats');
}
